src/CrossmintEmbed.ts

```diff
@@ -0,0 +1,225 @@
+import WindowAdapter from "./adapters/WindowAdapter";
+import { ALLOWED_ORIGINS } from "./consts/origins";
+import { CrossmintEmbedConfig } from "./types";
+import {
+    CrossmintEmbedRequestAccountData,
+    CrossmintEmbedRequestType,
+    CrossmintEmbedSignMessageData,
+} from "./types/requests";
+import { htmlToElement } from "./utils/iframe";
+import buildSiteMetadata from "./utils/siteMetadata";
+import { sleep } from "./utils/sleep";
+
+export default class CrossmintEmbed {
+    private _config: CrossmintEmbedConfig;
+
+    private get _frameUrl() {
+        return this._config.environment + "/frame";
+    }
+
+    private constructor(config: CrossmintEmbedConfig) {
+        console.log("[crossmint-embed] Initialized embed with version:", config.libVersion);
+
+        this._config = config;
+    }
+
+    static async init(config: CrossmintEmbedConfig) {
+        const client = new CrossmintEmbed(config);
+
+        return client;
+    }
+
+    async login(): Promise<string | undefined | null> {
```

Comment:
why do we handle both undefined and null? D:

What's the difference in meaning between those two? Just undefined is fine right? same everywhere else